File: day5/.py
# import these modules if you need them:
from datetime import datetime, timezone
# import math
      
def time_until_take_off(from_time: str, take_off_time: str) -> int:
  def converter(elftime):
    elftime = elftime[:-3]

    [y, m, extra] =  elftime.split("*")
    [d, extratime] = extra.split("@")
    [hh, mm, ss] = extratime.split("|")

    return datetime(
        int(y),
        int(m),
        int(d),
        int(hh),
        int(mm),
        int(ss),
        tzinfo=timezone.utc
    )
  
  return int((converter(take_off_time) - converter(from_time)).total_seconds())

File: day05/.ts
type ElfDateTime =
  `${number}*${number}*${number}@${number}|${number}|${number} NP`

function timeUntilTakeOff(
  fromTime: ElfDateTime,
  takeOffTime: ElfDateTime
): number {
  function converter(elftime:ElfDateTime){
    let time = elftime.slice(0, -3)

    let [y, m, extra] = time.split("*")
    let [d, extratime] = extra.split("@")
    let [hh, mm, ss] = extratime.split("|")

    return new Date(
      Date.UTC(
        Number(y),
        Number(m) - 1,
        Number(d),
        Number(hh),
        Number(mm),
        Number(ss),
      )
    )
  }
  return (Math.floor((converter(takeOffTime) - converter(fromTime)) / 1000))
}